examples/Chainweb/Simulate/Contracts/SimplePayments.hs

```diff
@@ -88,111 +91,52 @@ simplePaymentsContractLoader adminKeyset = do
-
-createAccounts :: IO [([SomeKeyPair], Command Text)]
-createAccounts = traverse (createAccount . go) (words names)
-  where
-    go (x:xs) = toUpper x : map toLower xs
-    go _ = error "impossible"
-
-names :: String
-names = "mary elizabeth patricia jennifer linda barbara margaret susan dorothy jessica james john robert michael william david richard joseph charles thomas"
-
-newtype Account = Account
-  { getAccount :: String
-  } deriving (Eq, Show, Generic)
-
-accountNames :: [Account]
-accountNames = map (Account . go) (words names)
-  where
-    go (x:xs) = toUpper x : map toLower xs
-    go _ = error "impossible"
-
-instance Fake Account where
-  fake = elements accountNames
-
-newtype Amount = Amount
-  { getAmount :: Decimal
-  } deriving (Eq, Show, Generic)
-
-instance Fake Amount where
-  fake =
-    Amount <$>
-    (realFracToDecimal <$> fromRange (0, 10) <*>
-     (fromRange (lowerLimit, upperLimit) :: FGen Double))
-    where
-      lowerLimit = 0
-      upperLimit = 1000
-
-newtype Balance = Balance
-  { getBalance :: Decimal
-  } deriving (Eq, Show, Generic)
-
-instance Fake Balance where
-  fake = Balance . fromIntegral <$> fromRange (0, 100000 :: Integer)
-
-distinctPair :: (Fake a, Eq a) => FGen (a,a)
-distinctPair = fake >>= \a -> (,) a <$> suchThat fake (/= a)
-
-mkRandomSimplePaymentRequest :: [(Account, [SomeKeyPair])] -> IO (FGen SimplePaymentRequest)
+mkRandomSimplePaymentRequest :: [(Account, Maybe [SomeKeyPair])] -> IO (FGen SimplePaymentRequest)
 mkRandomSimplePaymentRequest kacts = do
   request <- randomRIO (0, 1 :: Int)
   case request of
-    0 -> return $ RequestGetBalance <$> fake
+    0 -> return $ SPRequestGetBalance <$> fake
     1 -> return $ do
         (from, to) <- distinctPair
-        RequestPay from to <$> fake
+        SPRequestPay from to <$> fake
     -- Lol, this might be used later. For now, this constructor will
     -- not be exercised.
     2 -> return $ do
            acct <- fake
            bal <- fake
-           case lookup acct kacts of
+           case join (lookup acct kacts) of
             Nothing -> error (errmsg ++ (getAccount acct) ++ " " ++ show (fst <$> kacts))
-            Just keyset -> return $ CreateAccount acct bal keyset
+            Just keyset -> return $ SPCreateAccount acct bal keyset
     _ -> error "mkRandomSimplePaymentRequest: error in case statement."
   where
     errmsg =
            "mkRandomSimplePaymentRequest: something went wrong."
            ++ " Cannot find account name"
 
-
 data SimplePaymentRequest
-  = RequestGetBalance Account
-  | RequestPay Account Account Amount
-  | CreateAccount Account Balance [SomeKeyPair]
-
-parens :: String -> String
-parens s = "(" ++ s ++ ")"
+  = SPRequestGetBalance Account
+  | SPRequestPay Account Account Amount
+  | SPCreateAccount Account Balance [SomeKeyPair]
 
 instance Show SimplePaymentRequest where
```

Comment:
Is the existential in `SomeKeyPair` the reason this isn't being auto-derived?  If so, we're using `PublicMeta` in lieu of `SomeKeyPair` in other places.  Could we do that here?

Reply:
This show instance has been taken out. Done.

Reply:
I've resolved this by removing the show instances.